### PR TITLE
cli: reorder rebuild-index so that records are indexed w/o errors

### PR DESCRIPTION
**Pull request by fenekku**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -153,11 +153,6 @@ def app_logout(self):
         """Create identity for the user."""
         assert logout_user()
 
-    @identity.deleter
-    def identity(self):
-        """Delete the user."""
-        self._identity = None
-
```
Comment (fenekku) [inline tests/conftest.py:160]: drive-by improvement. This is a repeat of the method on L143

File: invenio_rdm_records/cli.py [hunk 2]
```diff
@@ -257,24 +257,20 @@ def create_fixtures():
...
     subj_service.rebuild_index(identity=system_identity)
 
     click.secho("Reindexing affiliations...", fg="green")
-
     affs_service = current_service_registry.get("affiliations")
     affs_service.rebuild_index(identity=system_identity)
 
+    click.secho("Reindexing records and drafts...", fg="green")
+    rec_service = current_rdm_records.records_service
+    rec_service.rebuild_index(identity=system_identity)
```
Comment (fenekku) [inline invenio_rdm_records/cli.py:274]: Rdmrecords needed to be re-indexed last for me, since they depend on vocabularies, subjects, affiliations.

**Review by ppanero — APPROVED** [review]
